crates/forge/src/scarb.rs

```diff
@@ -179,12 +179,21 @@ pub fn paths_for_package(
         .ok_or_else(|| anyhow!("Failed to find metadata for package = {package}"))?;
 
     let package_path = package.root.clone();
-
     let lib_path = compilation_unit.target.source_path.clone();
 
     Ok((package_path, lib_path))
 }
 
+pub fn target_dir_for_package(package_path: &Utf8PathBuf) -> Result<Utf8PathBuf> {
+    let scarb_metadata = MetadataCommand::new().inherit_stderr().exec()?;
+
+    let target_dir = scarb_metadata
+        .target_dir
+        .unwrap_or_else(|| package_path.join("target"));
```

Comment:
I wonder in which case can the `target_dir` be None in scarb metadata. Perhaps it makes more sense to just bubble the error here instead of using a default?

Reply:
👋 It is an option for backward compatibility with older scarb versions only. I do not see an issue with using a default here, although this should be workspace root, not package root.